scripts/unix/download-tool.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+###
+# Copyright 2016 resin.io
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+set -u
+set -e
+
+function check_dep() {
+  if ! command -v $1 2>/dev/null 1>&2; then
+    echo "Dependency missing: $1" 1>&2
+    exit 1
+  fi
+}
+
+check_dep wget
+
+if command -v sha256sum 2>/dev/null 1>&2; then
+  SHA256SUM=sha256sum
+elif command -v shasum 2>/dev/null 1>&2; then
+  SHA256SUM="shasum -a 256"
+else
+  echo "Dependency missing: sha256sum or shasum" 1>&2
+  exit 1
+fi
+
+function usage() {
+  echo "Usage: $0"
+  echo ""
+  echo "Options"
+  echo ""
+  echo "    -u <url>"
+  echo "    -c <checksum>"
+  echo "    -x set execution permissions"
+  echo "    -o <output>"
+  exit 1
+}
+
+ARGV_URL=""
+ARGV_CHECKSUM=""
+ARGV_EXECUTION_PERMISSIONS=false
+ARGV_OUTPUT=""
+
+while getopts ":u:c:o:x" option; do
+  case $option in
+    u) ARGV_URL="$OPTARG" ;;
+    c) ARGV_CHECKSUM="$OPTARG" ;;
+    x) ARGV_EXECUTION_PERMISSIONS=true ;;
+    o) ARGV_OUTPUT="$OPTARG" ;;
+    *) usage ;;
+  esac
+done
+
+if [ -z "$ARGV_URL" ] || \
+   [ -z "$ARGV_CHECKSUM" ] || \
+   [ -z "$ARGV_OUTPUT" ]
+then
+  usage
+fi
+
+function checksum_matches() {
+  local file=$1
+  local hash=$2
+	test "$($SHA256SUM $file | cut -d ' ' -f1)" = "$hash"
+}
+
+if [ -e "$ARGV_OUTPUT" ] && checksum_matches "$ARGV_OUTPUT" "$ARGV_CHECKSUM"; then
+  echo "Re-using from cache"
+  exit 0
+fi
+
+wget --no-check-certificate "$ARGV_URL" -O "$ARGV_OUTPUT"
+
+if ! checksum_matches "$ARGV_OUTPUT" "$ARGV_CHECKSUM"; then
+  echo "Checksum mismatch" 1>&2
+  rm -f "$ARGV_OUTPUT"
+  exit 1
```

Comment:
I propose to not bother with it for the time being. We'll see if it comes up in the future.